lib/curly/presenter.rb

```diff
@@ -52,6 +52,10 @@ def initialize(context, options = {})
           end
         end
 
+        if value.respond_to?(:call)
+          value = value.call
```

Comment:
I did this instead:

```diff
       self.class.presented_names.each do |name|
         value = options.fetch(name) do
-          default_values.fetch(name) do
+          default = default_values.fetch(name) do
             raise ArgumentError.new("required identifier `#{name}` missing")
           end
-        end
 
-        if value.respond_to?(:call)
-          value = value.call
+          if default.respond_to?(:call)
+            context.instance_exec(name, &default)
+          else
+            default
+          end
         end
 
         instance_variable_set("@#{name}", value)
```

But the problem is, how should default values propagate to multiple presenter values?